### PR TITLE
ci: Fix edge ci with updates

### PR DESCRIPTION
**Pull request by danielhjacobs**
Supersedes #17347 and #17255.

Requesting review from @torokati44 as the one who always does the manual upgrades. See also https://github.com/webdriverio-community/node-edgedriver/issues/365#issuecomment-2269695346.

### CODE REVIEW

**Comment by danielhjacobs** [comment]
This was the result of `npm remove wdio-chromedriver-service wdio-edgedriver-service wdio-geckodriver-service`, `npm install --save-dev chromedriver edgedriver geckodriver` and `npm install webdriverio`

File: web/package.json [hunk 1]
```diff
@@ -37,16 +40,10 @@
...
-        "wdio-chromedriver-service": "^8.1.1",
-        "wdio-edgedriver-service": "^3.0.3",
-        "wdio-geckodriver-service": "^5.0.2",
-        "webdriverio": "^8.33.1",
+        "webdriverio": "^8.40.0",
         "webpack": "^5.93.0",
         "webpack-cli": "^5.1.4"
     },
-    "optionalDependencies": {
-        "chromedriver": "^127.0.1"
```
Comment (torokati44) [inline web/package.json:48]: This was optional before - should it stay that way...?

**Comment by christian-bromann** [comment]
> Followed by `npm install --save-dev chromedriver edgedriver geckodriver`

These are already dependency of `@wdio/utils` and don't need to be your projects dependencies 😉

File: web/package.json (hunk 1 above)
Comment (danielhjacobs) [reply web/package.json:48]: Maybe. `wdio-chromedriver-service`, `wdio-edgedriver-service`, and `wdio-geckodriver-service` were devDependencies so I wasn't sure which was more correct.

**Comment by torokati44** [comment]
Hmmm...

> `Error: Error: Failed to initilialise launcher service unknown: Error: Couldn't find plugin "chromedriver" service, neither as wdio scoped package "@wdio/chromedriver-service" nor as community package "wdio-chromedriver-service". Please make sure you have it installed!`

Also, look at the typo there, @christian-bromann: :sweat_smile: `initilialise`

**Comment by danielhjacobs** [comment]
Tried just `npm remove wdio-chromedriver-service wdio-edgedriver-service wdio-geckodriver-service` and `npm install webdriverio` without any manual changes to the package files.

**Comment by danielhjacobs** [comment]
> These are already dependency of `@wdio/utils` and don't need to be your projects dependencies 😉

Yeah, this doesn't seem to be the case, as @torokati44 noted.

**Comment by danielhjacobs** [comment]
Yeah, this just isn't working @christian-bromann.

**Comment by christian-bromann** [comment]
You probably still have the services defined in your `wdio.conf.ts/js` file, you will have to remove the references there too, otherwise WebdriverIO will try to import them.

**Comment by torokati44** [comment]
> You probably still have the services defined in your wdio.conf.ts/js file, you will have to remove the references there too, ...

Right...: https://github.com/ruffle-rs/ruffle/blob/a00d0e94992084b3b99dba9958391e9e78c2ee8e/web/packages/selfhosted/wdio.conf.ts#L19-62

**Comment by danielhjacobs** [comment]
Ok, fixed, but now wdio/utils's edgedriver is being used, and I guess that hasn't been updated: https://github.com/ruffle-rs/ruffle/actions/runs/10255193161/job/28371652458

**Comment by christian-bromann** [comment]
Please update your `package-lock.json` and ensure your `edgedriver` dependency is on `v5.6.1`

**Comment by torokati44** [comment]
Thanks for popping in to help, @christian-bromann, this worked!
Are you okay with merging this as is, and closing the two superseded PRs, @danielhjacobs?

**Comment by torokati44** [comment]
Thank you for really getting to the bottom of this!

**Pull request merged**